### PR TITLE
Bugfix for multiple routers under Riak

### PR DESCRIPTION
**Pull request by seancribbs**
The previous addition of multiple routers -- for the benefit of launching multiple applications -- assumed that any process calling `webmachine_mochiweb:start/1` would give its name as the name of a router process. However, in `riak_core`, the name given is:
1. A string, which is not a valid registered name.
2. Related to the interface/port on which the mochiweb listener binds, not the dispatch list.

This PR changes the behavior of multiple dispatch lists slightly such that:
1. There is only one `webmachine_router` process, but it has multiple dispatch lists in its ETS table. Since the table is public, we don't need to run multiple routers, just have a way to access the dispatch lists by name.
2. The desired dispatch list name is given as the `router_name` option, defaulting to `webmachine_router`. The `name` option given to `webmachine_mochiweb:start/1` is used ONLY as part of the listener process name, defaulting to `webmachine`.

### CODE REVIEW

**Comment by bookshelfdave** [comment]
This patch allows riak_ee 684409f8bb to startup. All tests pass.

:+1: from me, but probably needs a quick review from someone who knows the ins and outs of wm better than I do.

**Comment by reiddraper** [comment]
Please wait for us to touch bases re: how this affects Riak CS before merging.

**Comment by buddhisthead** [comment]
Just want to add my +1 here.
riak_test _anything_ crashes on master without this branch, and runs correctly with it.
It's a must-have unless we revert the prior commit.

**Comment by kellymclaughlin** [comment]
Hmm I was under the impression riak was pegged to a stable tag of
webmachine. Reverting the change is not really an option because we need it
for CS. I'll be happy to look further at this when I get a few minutes
unless somebody gets to it first.

**Comment by seancribbs** [comment]
@kellymclaughlin This does not require reverting anything, just adding another option to the option-list in CS, naming the "router" you want to use for each listener.

**Comment by reiddraper** [comment]
I'm happy to try this patch on CS this evening or weekend, but in the meantime please just update your Riak rebar.config to use a previous commit of webmachine.

File: src/webmachine_mochiweb.erl [hunk 1]
```diff
@@ -125,12 +107,17 @@ get_wm_options(Options) ->
...
-                {webmachine_router, Opts2};
-            Res ->
-                Res
+                {webmachine, Opts2};
+            NRes -> NRes
+        end,
+    {RName, Options3} =
+        case get_option(router_name, Options2) of
+            {undefined, Opts3} ->
+                {webmachine_router, Opts3};
```
Comment (kellymclaughlin) [inline src/webmachine_mochiweb.erl]: Maybe just me, but `webmachine_router` now seems like a misleading name. Would something like `dispatch_list` or `dispatch_group` be more indicative of what this is for?

Comment (kellymclaughlin) [inline src/webmachine_mochiweb.erl]: Oops sorry, i meant `router_name` and not `webmachine_router`.

Comment (seancribbs) [inline src/webmachine_mochiweb.erl]: @kellymclaughlin Yes, good point. Will change to `dispatch_list`, barring objections.

Comment (kellymclaughlin) [inline src/webmachine_mochiweb.erl]: +1

Comment (seancribbs) [inline src/webmachine_mochiweb.erl]: Actually, scratch that (post-workout-induced fuzziness), `dispatch_group` or `routes_group` makes more sense. Thoughts about the `webmachine_router` value for the default? I made it that way for compatibility but I suspect we could just use `default`.

Comment (kellymclaughlin) [inline src/webmachine_mochiweb.erl]: I think `default` works well.

Comment (reiddraper) [inline src/webmachine_mochiweb.erl]: +1 to `default` and either `dispatch_group` or `routes_group` (though there I slight prefer the singular, `route_group`)

**Comment by seancribbs** [comment]
Renames are done.

**Comment by kellymclaughlin** [comment]
Looks good, I really like this as opposed to the multiple router approach. +1 to merge.

**Pull request merged**